website/templates/profile/oauth_app_list.mako

```diff
@@ -0,0 +1,71 @@
+<%inherit file="base.mako"/>
+<%def name="title()">OAuth Application settings</%def>
+<%def name="content()">
+<h2 class="page-header">OAuth application settings</h2>
+
+<div class="row">
+
+    <div class="col-sm-3">
+        <div class="panel panel-default">
+            <ul class="nav nav-stacked nav-pills">
+                <li><a href="${ web_url_for('user_profile') }">Profile Information</a></li>
+                <li><a href="${ web_url_for('user_account') }">Account Settings</a></li>
+                <li><a href="${ web_url_for('user_addons') }">Configure Add-ons</a></li>
+                <li><a href="${ web_url_for('user_notifications') }">Notifications</a></li>
+                <li><a href="#">Developer apps</a> </li>
+            </ul>
+        </div><!-- end sidebar -->
+    </div>
+
+    <div class="col-sm-9 col-md-7">
+        <a href="${web_url_for('oauth_application_register')}" role="button" class="btn btn-default pull-right"><i class="fa fa-plus"></i> Register new application</a>
+        <div id="app-list">
+
+            <p data-bind="visible: (content().length == 0)">You have not registered any applications that can connect to the OSF.</p>
```

Comment:
I'm a little concerned, in general, about the usability of this feature. So, on the one hand, it's a developer-specific feature, and it's being implemented in the way that it is everywhere else, so that's good. On the other, if you don't know what it is, will it get confused with Addons? We might want some help text to explain to people under what circumstances they would use this feature and try to make it super-clear that this isn't for the sorts of things that addons do. But without being really text heavy, 'cause nobody reads a bunch of text.